libgalois/src/PropertyGraph.cpp

```diff
@@ -530,3 +530,209 @@ katana::SortNodesByDegree(katana::PropertyGraph* pg) {
 
   return katana::ResultSuccess();
 }
+
+katana::Result<std::unique_ptr<katana::PropertyGraph>>
+katana::CreateSymmetricGraph(katana::PropertyGraph* pg) {
+  const GraphTopology& topology = pg->topology();
+  auto symmetric = std::make_unique<katana::PropertyGraph>();
+  if (topology.num_nodes() == 0) {
+    return std::unique_ptr<PropertyGraph>(std::move(symmetric));
+  }
+
+  // New symmetric graph topology
+  auto out_indices = std::make_unique<katana::LargeArray<uint64_t>>();
+  auto out_dests = std::make_unique<katana::LargeArray<uint32_t>>();
+  katana::LargeArray<uint64_t> out_dests_offset;
+
+  out_indices->allocateInterleaved(topology.num_nodes());
+  out_dests_offset.allocateInterleaved(topology.num_nodes());
+
+  auto edges = topology.edges(0);
+  (*out_indices)[0] = *edges.begin();
```

Comment:
I'm surprised `0` index is handled in a special way. Doesn't `(*edges.end() - *edges.begin())` work?

Reply:
Good point. Thanks.